JitsiConference.js

```diff
@@ -371,13 +374,31 @@ JitsiConference.prototype._init = function(options = {}) {
     this.eventManager.setupStatisticsListeners();
 
     if (config.enableTalkWhileMuted) {
-        // eslint-disable-next-line no-new
-        new TalkMutedDetection(
-            this,
-            () =>
+        // If VAD processor factory method is provided uses VAD based detection, otherwise fallback to audio level
+        // based detection.
+        if (config.createVADProcessor) {
+            logger.info('Using VAD detection for generating talk while muted events');
+            // eslint-disable-next-line no-new
+            this._talkWhileMutedDetection = new VADTalkMutedDetection(this, config.createVADProcessor);
+            this._talkWhileMutedDetection.on(DetectionEvents.VAD_TALK_WHILE_MUTED, () =>
                 this.eventEmitter.emit(JitsiConferenceEvents.TALK_WHILE_MUTED));
+
+        } else {
+            logger.info('Using audio level based detection for generating talk while muted events');
+            this._talkWhileMutedDetection = new TalkMutedDetection(
+                this, () => this.eventEmitter.emit(JitsiConferenceEvents.TALK_WHILE_MUTED));
+        }
     }
 
+    // Generates events based on no audio input detector.
+    // eslint-disable-next-line
```

Comment:
Why?

Reply:
Same as above (forgot about it). Removed now.